### PR TITLE
Implemented userService.getUserByAuthId()

### LINKED ISSUE #119: Implement userService.getUserByAuthId()
## Description
 
Implement a service function to retrieve one user from the db using an `auth_id`. Only the service function is needed (not an endpoint).

## Acceptance Criteria

- [x] Create a function called `getUserByAuthId()` for the `user.service.ts` file

### PR DESCRIPTION
**Pull request by jimwang6012**
Fixes #119 

Created a function called `getUserByAuthId()` in the `user.service.ts` file. This function takes in a `uid` as a parameter and returns a user in the format specified in `UserModel` in `user.model.ts`. The `uid` parameter can be retrieved from the `decodedToken` object within the authorization header of the incoming request, that is `req.headers.authorization`. 

More details on this can be found here #123

### CODE REVIEW

**Review by sdiv877 — APPROVED** [review]
LGTM!

**Review by lanxdrey19 — APPROVED** [review]

File: backend/src/services/user.service.ts [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
-    await user.save();
-    return user;
+  const user = new User(userDetails);
+  await user.save();
+  return user;
+};
+
+export const getUserByAuthId = async (uid) => {
+  const user = await User.findOne({ auth_id: uid });
+  return user;
```
Comment (AdamAWiener) [inline backend/src/services/user.service.ts:11]: What happens when we dont find a user with that auth_id? What gets returned?

Comment (sdiv877) [reply backend/src/services/user.service.ts:11]: `null` would get returned and there would have to be a `if (user != null)` check inside whichever controller requests the user I think

**Review by UoA-sjer215 — APPROVED** [review]
lgtm

**Review by hajineats — APPROVED** [review]
LGTM

**Pull request merged**